src/inertia.js

```diff
@@ -61,7 +61,7 @@ export default {
       data: method.toLowerCase() === 'get' ? {} : data,
       params: method.toLowerCase() === 'get' ? data : {},
       cancelToken: this.cancelToken.token,
-      onUploadProgress,
+      ...config,
```

Comment:
I refactored this to let the user add any of the [axios config options](https://github.com/axios/axios#request-config).

Usage now:
```javascript
Inertia.post(this.route('files.store'), data, {
  config: {
     onUploadProgress: e => {
      const percentCompleted = Math.round((e.loaded * 100) / e.total);
      console.log(`${percentCompleted}% uploaded`);
    },
  },
})
```